### PR TITLE
[Filebeat] Add PanOS Global Protect & User ID logs

### LINKED ISSUE #24722: [Filebeat] Palo Alto integration with User-ID
Hi, everyone

I have been working with **Palo Alto** and **Filebeat** over several days. I have looked on [Elastic documentation](https://www.elastic.co/guide/en/beats/filebeat/current/filebeat-module-panw.html) that it currently supports messages of **Traffic** and **Threat** types.

Is it considered to parsing [User-ID](https://docs.paloaltonetworks.com/pan-os/9-1/pan-os-admin/monitoring/use-syslog-for-monitoring/syslog-field-descriptions/user-id-log-fields.html) type ? I would like to get it because it provides information about **login** and **logouts** of **usernames**.

Thanks in advance,

Rodrigo

### LINKED ISSUE #24724: [Filebeat] Palo Alto integration with GlobalProtect
Please provide support for the panw panos globalprotect log type.. 

Example logs:

`1,2021/03/24 11:30:00,022201001305,GLOBALPROTECT,0,2305,2021/03/24 11:30:00,vsys1,portal-prelogin,before-login,,,,BE,,11.134.5.168,0.0.0.0,0.0.0.0,0.0.0.0,09300bcc-23-4900-8de9-32695452fa,,5.2.4,Windows,"Microsoft Windows 10 Pro , 64-bit",1,,,"",success,,0,,0,GlobalProtect Portal,69200719497738,0x0`

`1,2021/03/24 11:29:49,022201001308,GLOBALPROTECT,0,2305,2021/03/24 11:29:49,vsys1,gateway-config-release,configuration,,,domain\user,BE,CP935,83.14.113.11,0.0.0.0,10.20.13.217,0.0.0.0,e0957c11-93-437a-9e23-9f0c24059898,5J9VN53,5.2.4,Windows,"Microsoft Windows 10 Pro , 64-bit",1,,,"",success,,0,,0,GlobalProtect_GW,6919501582016786,0x0`

### PR DESCRIPTION
**Pull request by legoguy1000**
## What does this PR do?

Updates the Panw PanOS module to parse the Palo Alto Global Protect and User ID logs.

## Why is it important?

Currently Global Protect and User ID logs are not parsed.

## Checklist

- [X] My code follows the style guidelines of this project
- [X] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have made corresponding change to the default configuration files
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added an entry in `CHANGELOG.next.asciidoc` or `CHANGELOG-developer.next.asciidoc`.

## Related issues

- Closes #24722 
- Closes #24724

### CODE REVIEW

File: x-pack/filebeat/module/panw/panos/ingest/pipeline.yml [hunk 1]
```diff
@@ -1,616 +1,530 @@
...
+      if: "ctx?.panw?.panos?.action != null && ['alert', 'allow', 'continue'].contains(ctx.panw.panos.action)"
+  - append:
+      field: event.type
+      allow_duplicates: false
+      value: denied
+      if: "ctx?.panw?.panos?.action != null && ['deny', 'drop', 'reset-client', 'reset-server', 'reset-both', 'block-url', 'block-ip', 'random-drop', 'sinkhole', 'block'].contains(ctx.panw.panos.action)"
+  - set:
+      field: event.outcome
+      value: failure
+      if: "ctx?.event?.type != null && ctx?.event?.type.contains('denied')"
```
Comment (adriansr) [inline x-pack/filebeat/module/panw/panos/ingest/pipeline.yml:181]: This is causing a lot of events in the traffic logs to be marked as `failure`. The way I see it, the event types set by the processor above should always be `success`, as in, the firewall was successful in denying a connection, dropping traffic or blocking an URL.

Comment (legoguy1000) [reply x-pack/filebeat/module/panw/panos/ingest/pipeline.yml:181]: Copy that, @leehinman and I had talked about that in another PR. I'll update that.

File: x-pack/filebeat/module/panw/panos/_meta/fields.yml [hunk 2]
```diff
@@ -147,3 +147,162 @@
...
+    # - name: hostid
+    #   type: keyword
+    #   description: >
+    #     The unique ID that GlobalProtect assigns to identify the host.
+
+    # - name: machinename
+    #   type: keyword
+    #   description: >
+    #     The name of the user’s machine.
+
```
[the remark this answers is not in the document]
Comment (legoguy1000) [reply x-pack/filebeat/module/panw/panos/_meta/fields.yml:165]: done

File: x-pack/filebeat/module/panw/panos/ingest/pipeline.yml (hunk 1 above)
Comment (legoguy1000) [reply x-pack/filebeat/module/panw/panos/ingest/pipeline.yml:181]: done

**Comment by legoguy1000** [comment]
@adriansr Should be GTG now.

**Review by adriansr — APPROVED** [review]
LGTM

**Pull request merged**